src/core/server/services/perspective/index.ts

```diff
@@ -0,0 +1,146 @@
+import { Db } from "mongodb";
+
+import { TOXICITY_ENDPOINT_DEFAULT } from "coral-common/constants";
+import logger from "coral-server/logger";
+import { Comment } from "coral-server/models/comment";
+import { findStory, getURLWithCommentID } from "coral-server/models/story";
+
+import {
+  GQLCOMMENT_STATUS,
+  GQLPerspectiveExternalIntegration,
+} from "coral-server/graph/tenant/schema/__generated__/types";
+
+interface SendResult {
+  ok: boolean;
+  status: number;
+  data: any;
+}
+
+async function send(
+  endpoint: string,
+  apiKey: string,
+  method: string,
+  body: any
+): Promise<SendResult> {
+  const result = await fetch(`${endpoint}/${method}?key=${apiKey}`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(body, null, 2),
+  });
+
+  if (!result.ok) {
+    return {
+      ok: result.ok,
+      status: result.status,
+      data: null,
+    };
+  }
+
+  const data = await result.json();
+
+  return {
+    ok: result.ok,
+    status: result.status,
+    data,
+  };
+}
+
+function computeStatus(status: GQLCOMMENT_STATUS) {
+  if (status === GQLCOMMENT_STATUS.APPROVED) {
+    return "APPROVED";
+  }
+  if (status === GQLCOMMENT_STATUS.REJECTED) {
+    return "DELETED";
+  }
+
+  return null;
+}
+
+export async function notifyPerspectiveModerationDecision(
+  mongo: Db,
+  domain: string,
+  config: GQLPerspectiveExternalIntegration,
+  comment: Comment,
+  status: GQLCOMMENT_STATUS
+) {
+  if (!config.enabled) {
+    return;
+  }
+  if (!config.key) {
+    return;
+  }
+  if (!config.sendFeedback) {
+    return;
+  }
+
+  const endpoint = config.endpoint
+    ? config.endpoint
+    : TOXICITY_ENDPOINT_DEFAULT;
+  const apiKey = config.key;
+
+  const communityId = `Coral:${domain}`;
+  const clientToken = `comment:${comment.id}`;
+  const latestRevision = comment.revisions[comment.revisions.length - 1];
```

Comment:
The latest revision in this case should not be used. Because approve and reject requires you to specify the revision ID being approved/rejected, that revision ID should be used to extract the correct revision to send to perspective instead.